cmd/influxd/upgrade/config.go

```diff
@@ -45,12 +61,22 @@ var configMapRules = map[string]string{
 	"http.https-private-key":                               "tls-key",
 }
 
+// configValueTransforms is a map from 2.x config keys to transformation functions
+// that should run on the 1.x values before they're written into the 2.x config.
+//
+// NOTE: This might be overkill for only 1 key, I wrote this before we had fully
+// investigated which config keys needed to be transformed in this way.
+var configValueTransforms = map[string]func(interface{}) interface{}{
+	"query-concurrency": rewriteInf,
```

Comment:
The patch here is definitely overkill if we only ever end up with 1 entry in this map. I wrote the fix before we'd nailed down the full list of settings that needed to be transformed during upgrade. As far as I can tell, query-concurrency turned out to be the only one that actually needs it.

If we'd rather have a simpler patch I can replace all this with a hard-coded check for query-concurrency and 0 within the `updateV2Config` loop.